File: w11/d01/Kristen_Tonga/node_bananas/monkeysController.js
var Monkey = require(path.join(__dirname, 'monkey'));

exports.index = function(req, res){
  Monkey.find(function(err, monkeys){
    if (err) return console.log(err);
    res.send(monkeys);
  });
};

exports.create = function(req, res){
  Monkey.create({ name: req.body.name }, function(err, monkey){
    if (err) return console.log(err);
    res.send(monkey);
  });
};

exports.update = function(req, res){
 Monkey.findById(req.params.id, function(err, monkey){ $set: {name: req.body.name}}, function (err, monkey){
  if (err) return console.error(err);
  res.send(monkey);
 });
};

exports.delete = function(req, res){
  monkey = Monkey.findById(req.params.id, function(err, monkey){
    monkey.remove(function(err){
      if (err) return console.error(err);
      return res.send({});
    })
  })

};

File: w10/d04/Tim_Hannes/jungle_tree/app/controllers/monkeys_controller.rb
class MonkeysController < ApplicationController

  # monkeys GET    /monkeys(.:format)     monkeys#index
  #         POST   /monkeys(.:format)     monkeys#create
  #  monkey GET    /monkeys/:id(.:format) monkeys#show
  #         PATCH  /monkeys/:id(.:format) monkeys#update
  #         PUT    /monkeys/:id(.:format) monkeys#update
  #         DELETE /monkeys/:id(.:format) monkeys#destroy

  def index
    monkeys = Monkey.all
    render json: monkeys.to_json
    #don't care what they send back or ask for, we only accept/give json
  end

  def show
    monkey = Monkey.find(params[:id])
    render json: monkey.to_json
  end

  def create
    monkey = Monkey.create(monkey_params)
    render json: monkey.to_json
  end

  def update
    monkey = Monkey.find(params[:id])
    monkey.update(monkey_params)
    render json: monkey.to_json
  end

  def destroy
    monkey = Monkey.find(params[:id])
    monkey.destroy
    render json: monkey.to_json
  end

  private
  #private is not necessary, but we only want it available to this controller

  def monkey_params
    params.require(:monkey).permit(:name)
  end

end
